src/portfft/descriptor.hpp

```diff
@@ -491,18 +512,89 @@ class committed_descriptor {
     // Dummy parameter is needed as only partial specializations are allowed without specializing the containing class
     template <detail::level Lev, typename Dummy>
     struct inner {
-      static Scalar* execute(committed_descriptor& desc, dimension_struct& dimension_data);
+      static Scalar* execute(committed_descriptor& desc, std::vector<kernel_data_struct>& kernels);
     };
   };
 
   /**
    * Calculates twiddle factors for the implementation in use.
    *
-   * @param dimension_data data about the dimension for which twiddles are needed
+   * @param kernels vector of kernels
    * @return Scalar* USM pointer to the twiddle factors
    */
-  Scalar* calculate_twiddles(dimension_struct& dimension_data) {
-    return dispatch<calculate_twiddles_struct>(dimension_data.level, dimension_data);
+  Scalar* calculate_twiddles(detail::level level, std::vector<kernel_data_struct>& kernels) {
+    return dispatch<calculate_twiddles_struct>(level, kernels);
+  }
+
+  /**
+   * utility function to set specialization constants
+   * @tparam SubgroupSize Subgroup size
+   * @param top_level selected level of implementation
+   * @param prepared_vec vector as returned from prepare_implementation
+   * @param compute_direction direction of compute, as in forward or backward
+   * @param kernel_num which dimension are the kernels being built for
```

Comment:
This isn't very meaningful to me.

Reply:
which part in the above descriptions ?

Reply:
`kernel_num which dimension are the kernels being built for`

`kernel_num` sounds like a count, or an index. The `which dimension` makes it sounds like this is to with 1D, 2D, 3D etc. FFTs.

Reply:
I wanted to maintain the variable naming consistency. that's what `build_w_spec_const` calls it, and since that's being passed, I called it the same.
I could change it in both places

Reply:
Done

Reply:
You've changed it to *The dimension for which the kernels are being built* I think (or I'm now looking in the wrong place). That doesn't answer my above question above whether its a count or index or if it refers to the size of the input lengths vector. I still find this confusing.

Reply:
The doxygen reads as `@param dimension_num which dimension are the kernels being built for`
It is the count of the dimension, as in dimenion 0, dimension 1 ....

Reply:
I think I'm going to just accept I don't understand this!